homeassistant/components/home_connect/number.py

```diff
@@ -189,29 +192,49 @@ async def async_set_native_value(self, value: float) -> None:
                 },
             ) from err
 
-    async def async_fetch_constraints(self) -> None:
+    async def async_fetch_constraints(self, _: datetime | None = None) -> None:
         """Fetch the max and min values and step for the number entity."""
-        try:
-            data = await self.coordinator.client.get_setting(
-                self.appliance.info.ha_id, setting_key=SettingKey(self.bsh_key)
-            )
-        except HomeConnectError as err:
-            _LOGGER.error("An error occurred: %s", err)
-        else:
-            self.set_constraints(data)
+        setting_key = cast(SettingKey, self.bsh_key)
+        data = self.appliance.settings.get(setting_key)
+        if not data or not data.unit or not data.constraints:
+            try:
+                data = await self.coordinator.client.get_setting(
```

Comment:
Side note: I'm thinking we should add a lock to the coordinator, and that lock needs to be acquired before using the client to make requests. Ie only allow awaiting one request at a time.